### ISSUE TITLE
Order Details - state is shown incorrectly. Country is determined by Android framework.

### ISSUE THREAD

**Issue by wzieba** [woocommerce/woocommerce-android#5604]
E.g. state `Hokkaido` in `Japan` is shown as `JP01` which is location code for this state in WooCommerce Core.

Also, we format `country` based on `country code` via `java.util.Locale#displayCountry` (see: `AddressUtils`). As we have `Location` fetched from Core with its code and name, I believe we should prefer using that.

#### Core
<img width="681" alt="Screenshot 2022-01-06 at 15 31 40" src="https://user-images.githubusercontent.com/5845095/148399808-5807f13c-6d95-4785-ad5f-d1fcca1b01d7.png">

#### WooCommerce Android 8.1.1
<img width="721" alt="Screenshot 2022-01-06 at 15 31 49" src="https://user-images.githubusercontent.com/5845095/148399878-1a3b0fb3-1c6f-4ca0-b3f4-3f3723ba0e84.png">

#### `WCLocation`s from Core
<img width="1315" alt="Screenshot 2022-01-06 at 15 32 01" src="https://user-images.githubusercontent.com/5845095/148399931-6965d65a-7d72-4c2e-9ac0-518e8ab457b5.png">

**Comment by wzieba** [comment]
iOS seems to have similar issue:

<img src=https://user-images.githubusercontent.com/5845095/148400347-0e26e20a-875c-414e-b328-b8597ecab442.jpeg width=300>

**Comment by shiki** [comment]
This is currently marked as M1 but I think this shouldn't block the M1 release. What do you think? (cc @ThomazFB) 

Looks like iOS decided to do this in M2 too.

**Comment by ThomazFB** [comment]
@shiki I was waiting for us to finish this sprint and deliver the M1 itself before updating the pending issues to M2. But I agree with you, this should be tackled on M2.

**Comment by wzieba** [comment]
Duplicate of #2820 . Just FYI: I'm working on that

**Comment by wzieba** [comment]
Differences in address formatting on our platforms:

### Japan

| Core | Android | iOS |
| - | - | - |
| <img width="300" alt="Screenshot 2022-04-27 at 13 30 02" src="https://user-images.githubusercontent.com/5845095/165508989-9afc4775-3282-4626-8309-0b38b3a1a810.png"> | <img width="300" alt="Screenshot 2022-04-27 at 13 28 17" src="https://user-images.githubusercontent.com/5845095/165510083-1e4d9022-5517-49eb-9e85-ecce1a79abbf.png"> | <img width="300" src="https://user-images.githubusercontent.com/5845095/165509196-a47f2724-01ae-4602-a168-c642c766a1b7.PNG"> |

### US
| Core | Android | iOS |
| - | - | - |
| <img width="300" alt="Screenshot 2022-04-27 at 13 34 22" src="https://user-images.githubusercontent.com/5845095/165509523-9fbe9590-d58d-4f42-8366-6bc5fee60217.png"> | <img width="300" alt="Screenshot 2022-04-27 at 13 35 24" src="https://user-images.githubusercontent.com/5845095/165509655-67f602a2-41a2-49e7-abce-9d555972c362.png"> | <img width="300" src="https://user-images.githubusercontent.com/5845095/165509899-7d8bf4f3-46da-4bbb-9824-6f645b4defe4.jpeg"> |

**Comment by wzieba** [comment]
Internal discussion: p91TBi-8lv-p2

**Comment by nbradbury** [comment]
I added the "on hold" label to reflect that we're waiting on backend work, as per the internal discussion linked above.
